homeassistant/components/siren/__init__.py

```diff
@@ -64,10 +64,27 @@ def process_turn_on_params(
 
     if not supported_features & SUPPORT_TONES:
         params.pop(ATTR_TONE, None)
-    elif (tone := params.get(ATTR_TONE)) is not None and (
-        not siren.available_tones or tone not in siren.available_tones
-    ):
-        raise ValueError(f"Invalid tone received for entity {siren.entity_id}: {tone}")
+    elif (tone := params.get(ATTR_TONE)) is not None:
+        # Raise an exception if the specified tone isn't available
+        is_tone_dict_value = bool(
+            isinstance(siren.available_tones, dict)
+            and tone in siren.available_tones.values()
+        )
+        if not siren.available_tones or (
```

Comment:
Drop the last parenthesis too. Reading from left to right in one go is the easiest.

Reply:
I personally find it less readable without the parenthesis but maybe that's just me 🤷🏾